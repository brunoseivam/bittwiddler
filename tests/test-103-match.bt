main {
    var x : int16 = 0;

    match x {
        0 -> { emit("x is 0\n"); }
        _ -> { emit("x is not 0\n"); }
    };

    var month : int32 = 12;

    # monthName's type is inferred
    var monthName = match month {
        1  -> { "January";   }
        2  -> { "February";  }
        3  -> { "March";     }
        4  -> { "April";     }
        5  -> { "May";       }
        6  -> { "June";      }
        7  -> { "July";      }
        8  -> { "August";    }
        9  -> { "September"; }
        10 -> { "October";   }
        11 -> { "November";  }
        12 -> { "December";  }
        _  -> { "(INVALID)"; }
    };

    emit("Month #{month} is \"{monthName}\"\n");
}
