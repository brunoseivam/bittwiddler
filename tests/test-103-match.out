x is 0
Month #12 is "December"
